src/libecalc/application/graph_result.py

```diff
@@ -1,16 +1,50 @@
-from pydantic import BaseModel
+from datetime import datetime
 
+from pydantic import BaseModel, ConfigDict, Field
+
+from libecalc.common.utils.rates import TimeSeries
 from libecalc.common.variables import VariablesMap
 from libecalc.core.result import ComponentResult, EcalcModelResult
 from libecalc.core.result.emission import EmissionResult
 from libecalc.dto.component_graph import ComponentGraph
 
 
+def serialize_datetime(v):
+    if isinstance(v, datetime):
+        return v.strftime("%Y-%m-%d %H:%M:%S")
+    elif isinstance(v, list):
+        return [serialize_datetime(item) for item in v]
+    elif isinstance(v, dict):
+        return {str(k): serialize_datetime(vv) for k, vv in v.items()}
+    elif isinstance(v, BaseModel):
+        return {str(k): serialize_datetime(getattr(v, k)) for k in v.model_fields}
+    elif isinstance(v, TimeSeries):
+        return {
+            "periods": serialize_datetime(v.periods),
+            "values": serialize_datetime(v.values),
+            "unit": serialize_datetime(v.unit),
+        }
+    elif hasattr(v, "to_dict"):
+        return serialize_datetime(v.to_dict())
+    return v
```

Comment:
```suggestion
    return str(v)
```

isnt that what we want?

do we also want to catch other data types and log a warning in case there is some other data type?

Reply:
Have added log warnings if other types. And unit test to check log warnings.